### ISSUE TITLE
OpenDSS vs. GridLAB-D ieee123 model switch state inconsitency 

### ISSUE THREAD

**Issue by garydblack** [GRIDAPPSD/GOSS-GridAPPS-D#1313]
Per the file I attempted to attach, we are seeing an inconsistency in a switch state between our Y-bus query of OpenDSS and what we get from GridLAB-D at the first timestamp.  For the switch between nodes 61 and 61s for all phases 1, 2, and 3, the OpenDSS query indicates that the switch should be closed (value==1).  But the admittance given by the initial GridLAB-D measurement is low indicating the switch is actually open.

For the state estimator service we are going to be changing our implementation so the switch states are take solely from GridLAB-D as we need to implement this anyway to catch switch state changes during simulations, which is not currently supported.

No switch state inconsistencies were found for the 13-node model and the 9500-node model was not tested once we found this initial inconsistency (meaning we need to bump up the priority for determining our initial switch states from GridLAB-D measurements).

[sedbg.switch123.log](https://github.com/GRIDAPPSD/GOSS-GridAPPS-D/files/4920448/sedbg.switch123.log)

**Comment by garydblack** [comment]
I'm not sure if the file attachment worked, so here is the log file snippet with the inconsistent switch from state-estimator debug output:

measurement load_break_switch zid: _1b352927-624f-4418-a1d4-2eaa8d184815_switch
measurement load_break_switch value: 1
measurement load_break_switch znode1s: 61s.2
measurement load_break_switch znode2s: 61.2
measurement load_break_switch Ypu: (0,0)
measurement load_break_switch Yphys: (0,0)

measurement load_break_switch zid: _2754178b-3e50-4a4f-a26e-9f3e88e54674_switch
measurement load_break_switch value: 1
measurement load_break_switch znode1s: 61s.3
measurement load_break_switch znode2s: 61.3
measurement load_break_switch Ypu: (0,0)
measurement load_break_switch Yphys: (0,0)

measurement load_break_switch zid: _ff16b365-6d6b-445f-b4dd-1373f9dd8a3b_switch
measurement load_break_switch value: 1
measurement load_break_switch znode1s: 61s.1
measurement load_break_switch znode2s: 61.1
measurement load_break_switch Ypu: (0,0)
measurement load_break_switch Yphys: (0,0)

**Comment by garydblack** [comment]
Also note that for the same 123-node model, the switch between 149 and 150r is also inconsistent between OpenDSS and GridLAB-D.  As with 61 and 61s switch, OpenDSS says the switch is closed while GridLAB-D indicates it is open.

**Comment by shpoudel** [comment]
I am not sure if this issue is still relevant. However, I thought of providing my input. Based on these inconsistencies and both being related to those switches which are kind of floating on the feeder and both being connected to a regulator or Xfmr, it looks like the Ybus calculation is giving Ypu as 0.

**Comment by areiman** [comment]
Can you elaborate on "both being related to those switches which are kind of floating on the feeder and both being connected to a regulator or Xfmr"? Does this explain the inconsistency?

**Comment by shpoudel** [comment]
Both the switches (150r-149) and (61-61s) are connected to a regulator or Xfmr on one of the sides. All other normally closed switches in the system are connected to the neighboring nodes with at least two distribution lines. I just suspect this is giving a Ypu for those two switches a very less value. Just my thought, I might be wrong. Does this make sense? If not, I can explain in the meeting.

**Comment by areiman** [comment]
I see the pattern you're describing. Do you see a causal link between that pattern and the zero-admittance terms as well? Either way, thanks for the lead.

**Comment by temcdrm** [comment]
How critical is this?  I have to start prioritizing my issues for some available time next week.

**Comment by areiman** [comment]
I would call it an important source of error but it doesn't break the app.

**Comment by garydblack** [comment]
Tom, it appears this inconsistency is now gone for the two switches in the 123-node model.  Did you make changes?

**Comment by temcdrm** [comment]
No, I haven't done anything to address this issue.

**Comment by temcdrm** [comment]
I think we need to make sure the problem has really gone away first.  Maybe Andy F or another developer has changed something in the bridge, but if not, problems don't usually disappear by themselves.

**Comment by garydblack** [comment]
I agree.  That's why I want to leave it open for now.  But, just wanted to let you know to not do anything further regarding this until we look into it further.  I'll send an email to Andy F. and see if he knows what might have changed.

**Comment by garydblack** [comment]
We are now unable to reproduce this issue for the 123-node model.  Below is a screen capture showing the current switch state and admittance value, all of which are now consistent.  We have also checked the 9500-node model and verified there are no mismatches between switch state and admittance values.  We've decided to move this to the backlog for now and will revive if the problem reappears.

![image](https://user-images.githubusercontent.com/26394029/92979117-89805b00-f446-11ea-8f5b-3b303d95cc03.png)

**Comment by temcdrm** [comment]
It's been more than a year since the last report. I'm working on CIMHub now, and verified that both OpenDSS and GridLAB-D models are exported with those switches initially closed.
